dipy/denoise/denspeed.pyx

```diff
@@ -154,16 +158,19 @@ cdef double process_block(double [:, :, ::1] arr,
             for o in range(P, BS - P):
 
                 summ = 0
+                sigm = 0
 
                 # calculate square distance
-                for a in range(- P, P + 1):
-                    for b in range(- P, P + 1):
-                        for c in range(- P, P + 1):
+                for a in range(-P, P + 1):
+                    for b in range(-P, P + 1):
+                        for c in range(-P, P + 1):
 
                             # this line takes most of the time! mem access
                             d = cache[(B + a) * BS * BS + (B + b) * BS + (B + c)] - cache[(m + a) * BS * BS + (n + b) * BS + (o + c)]
                             summ += d * d
+                            sigm += sigma_block[(m + a) * BS * BS + (n + b) * BS + (o + c)]
```

Comment:
Ok, the compiler must already do that trivial optimization then. Thanks for testing it.